integrations/docker/images/chip-cert-bins/Dockerfile

```diff
@@ -0,0 +1,273 @@
+            && wget --progress=dot:giga https://github.com/Kitware/CMake/releases/download/v3.19.3/cmake-3.19.3-Linux-aarch64.sh \
+            && sh cmake-3.19.3-Linux-aarch64.sh --exclude-subdir --prefix=/usr/local \
+            && rm -rf cmake-3.19.3-Linux-aarch64.sh) \
+            && exec bash \
+            ;; \
+        *) \
+	    test -n "$TARGETPLATFORM" \
+	    echo "Unsupported platform ${TARGETPLATFORM}" \
+	;; \
+    esac
+
+# Python 3.9 and PIP
+RUN set -x \
+    && DEBIAN_FRONTEND=noninteractive  apt-get update \
+    && DEBIAN_FRONTEND=noninteractive  apt-get install -y libgirepository1.0-dev \
+    && DEBIAN_FRONTEND=noninteractive apt-get install -y software-properties-common \
+    && add-apt-repository universe \
+    && curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py \
+    && python3.9 get-pip.py \
+    && update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.9 1 \
+    && rm -rf /var/lib/apt/lists/ \
+    && : # last line
+
+RUN set -x \
+    && pip3 install attrs coloredlogs PyGithub pygit future portpicker mobly click cxxfilt ghapi pandas tabulate \
+    && : # last line
+
+# build and install gn
+RUN set -x \
+    && git clone https://gn.googlesource.com/gn \
+    && cd gn \
+    && python3 build/gen.py \
+    && ninja -C out \
+    && cp out/gn /usr/local/bin \
+    && cd .. \
+    && rm -rf gn \
+    && : # last line
+
+# Install bloat comparison tools
+RUN set -x \
+    && git clone https://github.com/google/bloaty.git \
+    && mkdir -p bloaty/build \
+    && cd bloaty/build \
+    && cmake ../ \
+    && make -j8 \
+    && make install \
+    && cd ../.. \
+    && rm -rf bloaty \
+    && : # last line
+
+# NodeJS: install a newer version than what apt-get would read
+# This installs the latest LTS version of nodejs
+RUN case ${TARGETPLATFORM} in \
+        "linux/amd64") \
+            set -x \
+			&& mkdir node_js \
+			&& cd node_js \
+			&& wget https://nodejs.org/dist/v12.19.0/node-v12.19.0-linux-x64.tar.xz \
+			&& tar xfvJ node-v12.19.0-linux-x64.tar.xz \
+			&& mv node-v12.19.0-linux-x64 /opt/ \
+			&& ln -s /opt/node-v12.19.0-linux-x64 /opt/node \
+			&& ln -s /opt/node/bin/* /usr/bin \
+			&& cd .. \
+			&& rm -rf node_js \
+			;; \
+        "linux/arm64")\
+            set -x \
+			&& mkdir node_js \
+			&& cd node_js \
+			&& wget https://nodejs.org/dist/v12.19.0/node-v12.19.0-linux-arm64.tar.xz \
+			&& tar xfvJ node-v12.19.0-linux-arm64.tar.xz \
+			&& mv node-v12.19.0-linux-arm64 /opt/ \
+			&& ln -s /opt/node-v12.19.0-linux-arm64 /opt/node \
+			&& ln -s /opt/node/bin/* /usr/bin \
+			&& cd .. \
+			&& rm -rf node_js \
+			;; \
+        *) ;; \
+    esac
+
+# Stage 1.5: Bootstrap Matter.
+# Set python to python3
+RUN update-alternatives --install /usr/bin/python python /usr/bin/python3 10
+
+RUN mkdir /root/connectedhomeip
+RUN git clone --depth 1 --single-branch --branch ${BRANCH} https://github.com/project-chip/connectedhomeip.git /root/connectedhomeip
```

Comment:
This pulls "latest version on branch" of this image, so the builds are not reproducible (me running today and running a few months from now do not produce the same output).

Reply:
Agreed, @andy31415 do you prefer using the SHA directly instead?

Reply:
If you use a tag here (eg. TE9), it would produce stable builds.

Also, since the intent is to upload these to docker hub, the images can be tagged with the commit hash and/or the branch/tag.

Reply:
I guess I would like to know the use case to specify a specific hash for the test harness images. AFAIK we always use the latest of X branch for test harness images. Doing this will require cloning the entire repo which I'm trying to avoid if I can.

Reply:
I have done this on a different project:

```sh
git clone --filter=blob:none --no-checkout --depth 1 https://github.com/project-chip/connectedhomeip.git 
cd connectedhomeip
git checkout -q sha
```

could that work here?

Reply:
I think @andy31415 want's to be able to reproduce the build later in time.
This is not possible if the branch moved ahead in the mean time.

Reply:
Hmm, but why would we want to do that if the images are uploaded to docker hub?

Another downside of attaching to a commit is if the target branch/tag updates, you have to submit another PR to update the Dockerfile.

How will Matter SDK releases and Test Harness releases be handled moving forward? My understanding is Test Harness is always attached to a specific branch or tag, which the current solution handles. If moving forward we will test and build the TH using specific commits instead, I can change the Dockerfile to accept a commit hash when building and not provide a default. This can be passed in when building from cli or from a Github action if we want to create one later.

Reply:
I'm OK going with branch for now. I think tag'ing the published images with SHA is the most important

Reply:
@davidgoogle would it be possible to have a `sdk_source` variable here, so instead of doing a remote clone, it just copies the local sdk checkout into the container?

Maybe we can move this to an Issue for enhancements. Here's how I think it could work.

```docker
ARG sdk_source="remote"

# Stage 1: Setup dependencies (based on chip-build).
FROM ubuntu:focal as chip-build-cert
<Same as you have line 3-188>

# Stage 2-Remote: Remote source from GitHub
FROM chip-build-cert AS chip-build-cert-source-remote

RUN mkdir /root/connectedhomeip
RUN git clone --depth 1 --single-branch --branch ${BRANCH} https://github.com/project-chip/connectedhomeip.git /root/connectedhomeip

# Stage 2-Local: Local source copied in
FROM chip-build-cert AS chip-build-cert-source-local

RUN mkdir /root/connectedhomeip
COPY <local_path> /root/connectedhomeip

# Stage 3: Build
FROM chip-build-cert-source-${sdk_source} as chip-build-cert-bins
SHELL ["/bin/bash", "-c"]
RUN scripts/build/gn_bootstrap.sh
RUN gn gen out/debug --args='chip_mdns="platform" chip_inet_config_enable_ipv4=false'
RUN ninja -C out/debug

<same as yours from line 200 onwards>
```

Reply:
That is how it worked in the previous PR, but @andy31415 was against copying the local checkout.

Reply:
Changed to accept a specific commit hash.